### DISCUSSION TITLE
Curved Waveguide

### DISCUSSION THREAD

**Discussion by yaiza-c** [NanoComp/meep#2835] [Q&A]
Hii

I'm trying to simulate a waveguide on meep like the one of the image. From the side it follows a Bézier curve and the cross section is an square of 2x2um. I need it to be a 3D simulation and, as far as I know, there isn't a way to define a curve in the geometry.

Any ideas?

![Screenshot from 2024-05-15 17-41-13](https://github.com/NanoComp/meep/assets/169924409/f1c717ae-6e78-4eee-9074-78bfac2f3209)

**Comment by stevengj** [comment]
You can define this in various ways.  One way is to use a CAD program to export the curve to a GDSII file, which can be [imported into Meep](https://meep.readthedocs.io/en/latest/Python_Tutorials/GDSII_Import/).  Another way is to discretize the curve yourself into a sequence of points (a polygon), which you then input via the [`meep.Prism` type](https://meep.readthedocs.io/en/latest/Python_User_Interface/#prism) as an "extruded" 3d object.   Yet another way is to define an external file holding the materials in whatever shape you want, using whatever software you want, and import it via `epsilon_input_file`.  Another way is to input an arbitrary ε(x,y,z) function in Python as an `epsilon_func` parameter.  Another way is to define a 2d "image" of the cross-section you want, and "extrude" it into 3d using a [`meep.MaterialGrid`](https://meep.readthedocs.io/en/latest/Python_User_Interface/#materialgrid).

I'm guessing either a `Prism` or GDSII file will be easiest for you.
